main/lsp/tools/make_lsp_types.cc

```diff
@@ -1284,6 +1277,12 @@ void makeLSPTypes(vector<shared_ptr<JSONClassType>> &enumTypes, vector<shared_pt
     // Empty object.
     auto InitializedParams = makeObject("InitializedParams", {}, classTypes);
 
+    auto DidChangeConfigurationParams = makeObject("DidChangeConfigurationParams",
+                                                   {
+                                                       makeField("settings", SorbetInitializationOptions),
```

Comment:
Curious what your thought process was for this API.

Another option would be to simply reuse SorbetInitializationOptions at the top-level of the didChangeConfiguration request, instead of having a `"settings"` field.

I'm not sure which I like more, as well as whether you considered other APIs before settling on this one.

Reply:
Just going according to the official spec https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_didChangeConfiguration 

but happy to do just top-level if we feel strongly about it

Reply:
oh whoops, I didn't realize the spec dictated that. This looks good!